subprojects/ide-native/src/integTest/groovy/org/gradle/ide/visualstudio/VisualStudioMultiProjectIntegrationTest.groovy

```diff
@@ -186,6 +194,101 @@ class VisualStudioMultiProjectIntegrationTest extends AbstractInstalledToolChain
         greetLibProject.projectConfigurations['debug'].includePath == filePath("src/greetings/headers")
     }
 
+    @RequiresInstalledToolChain(ToolChainRequirement.VISUALCPP_2017_OR_NEWER)
```

Comment:
I assume we don't use `ToolChainRequirement.VisualCPP` because we rely on `vswhere` to find msbuild.  Is there anything version-specific about how we are calling msbuild?  I wonder if we could also look in some known locations for older versions (i.e. `_VisualStudioInstallDir_/../MSBuild/_VisualCppVersion_/bin/MSBuild.exe`) and change this to `ToolChainRequirement.MSBuild`.  Then for each `InstalledToolChain` we check for the "known" location of MSBuild in the `meets()` method based on the version of VisualStudio the toolchain represents.

It would be nice to be able to validate that we haven't introduced something VS2017-specific in the metadata by checking against older versions of MSBuild.

Either way, it feels like a `ToolChainRequirement.MSBuild` better reflects the requirement, even if we only return true for VS2017.

Reply:
Good point, let me see what I can do.

Reply:
Does a different version of msbuild come with each VS installation?  If so and our assumption is that msbuild ~= VS behavior, I think it's important that these tests use each version of msbuild so we can be somewhat confident that each VS version works.  

I can't tell if we have a similar problem as this: https://github.com/gradle/gradle-native/issues/393  At this time, I don't think we need to test the combination of VS with other tools (e.g., msbuild from VS2017 building a project with GCC), but if we have the same problem as Xcode, we should open an issue for it.

Reply:
Done.